workers/unity/Packages/io.improbable.gdk.buildsystem/Tests/BuildContextTests.cs

```diff
@@ -0,0 +1,135 @@
+        [TestCase("UnityGameLogic", 1, BuildTarget.StandaloneWindows64)]
+        [TestCase("DummyWorkerType", 0, BuildTarget.StandaloneWindows64)]
+#elif UNITY_EDITOR_OSX
+        [TestCase("UnityClient", 1, BuildTarget.StandaloneOSX)]
+        [TestCase("UnityGameLogic", 1, BuildTarget.StandaloneOSX)]
+        [TestCase("DummyWorkerType", 0, BuildTarget.StandaloneOSX)]
+#endif
+        [TestCase("UnityClient", 0, BuildTarget.StandaloneLinux64)]
+        [TestCase("UnityGameLogic", 0, BuildTarget.StandaloneLinux64)]
+        [TestCase("DummyWorkerType", 0, BuildTarget.StandaloneLinux64)]
+        public void GetBuildContexts_for_worker_type_with_filter_returns_expected_number_of_local_contexts(string workerType, int expectedContexts, params BuildTarget[] buildTargetFilter)
+        {
+            var buildContextFilter = BuildContextFilter.Local(workerType);
+            buildContextFilter.BuildTargetFilter = buildTargetFilter;
+
+            List<BuildContext> buildContexts = null;
+            Assert.DoesNotThrow(() =>
+            {
+                buildContexts = BuildContext.GetBuildContexts(buildContextFilter);
+            });
+
+            Assert.AreEqual(expectedContexts, buildContexts.Count);
+        }
+
+        [TestCase("UnityClient", 2, BuildTarget.StandaloneWindows64, BuildTarget.StandaloneOSX)]
+        [TestCase("UnityClient", 1, BuildTarget.StandaloneWindows64, BuildTarget.StandaloneLinux64)]
+        [TestCase("UnityClient", 0, BuildTarget.StandaloneLinux64)]
+        [TestCase("UnityGameLogic", 1, BuildTarget.StandaloneLinux64)]
+        [TestCase("UnityGameLogic", 0, BuildTarget.StandaloneWindows64)]
+        [TestCase("DummyWorkerType", 0, BuildTarget.StandaloneWindows64, BuildTarget.StandaloneLinux64)]
+        [TestCase("DummyWorkerType", 0, BuildTarget.StandaloneLinux64)]
+        public void GetBuildContexts_for_worker_type_with_filter_returns_expected_number_of_cloud_contexts(string workerType, int expectedContexts, params BuildTarget[] buildTargetFilter)
+        {
+            var buildContextFilter = BuildContextFilter.Cloud(workerType);
+            buildContextFilter.BuildTargetFilter = buildTargetFilter;
+
+            List<BuildContext> buildContexts = null;
+            Assert.DoesNotThrow(() =>
+            {
+                buildContexts = BuildContext.GetBuildContexts(buildContextFilter);
+            });
+
+            Assert.AreEqual(expectedContexts, buildContexts.Count);
+        }
+
+#if UNITY_EDITOR_WIN
+        [TestCase("UnityClient", BuildTarget.StandaloneWindows64)]
+        [TestCase("UnityGameLogic", BuildTarget.StandaloneWindows64)]
+#elif UNITY_EDITOR_OSX
+        [TestCase("UnityClient", BuildTarget.StandaloneOSX)]
+        [TestCase("UnityGameLogic", BuildTarget.StandaloneOSX)]
+#endif
+        public void GetBuildContexts_for_valid_worker_type_without_filter_returns_all_expected_local_contexts(string workerType, params BuildTarget[] expectedTargets)
```

Comment:
The `expectedTargets` parameter isn't used in the body of the test

Reply:
Ah yes I forgot something during a copypasta